cmd/kube-rbac-proxy/app/kube-rbac-proxy.go

```diff
@@ -272,38 +271,20 @@ func Run(cfg *server.KubeRBACProxyConfig) error {
 
 // setupAuthorizer runs different authorization checks based on the configuration.
 func setupAuthorizer(krbInfo *server.KubeRBACProxyInfo, delegatedAuthz *serverconfig.AuthorizationInfo) (authorizer.Authorizer, error) {
-	// pathAuthorizer is running before any other authorizer.
-	// It works outside of the default authorizers.
-	var pathAuthorizer authorizer.Authorizer
-	var err error
-	// AllowPaths are the only paths that are not denied.
-	// IgnorePaths bypass all authorization checks.
-	switch {
-	case len(krbInfo.AllowPaths) > 0:
-		pathAuthorizer, err = path.NewAllowedPathsAuthorizer(krbInfo.AllowPaths)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create allow path authorizer: %w", err)
-		}
-	case len(krbInfo.IgnorePaths) > 0:
-		pathAuthorizer, err = path.NewPassthroughAuthorizer(krbInfo.IgnorePaths)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create ignore path authorizer: %w", err)
-		}
-	}
-
-	// Delegated authorizers are running after the pathAuthorizer
+	// authz are running after the pathAuthorizer
 	// and after the attributes have been rewritten.
-	var delegated []authorizer.Authorizer
+	// Default k8s authorizer
+	authz := delegatedAuthz.Authorizer
+
 	// Static authorization authorizes against a static file is ran before the SubjectAccessReview.
 	if krbInfo.Authorization.Static != nil {
 		staticAuthorizer, err := static.NewStaticAuthorizer(krbInfo.Authorization.Static)
 		if err != nil {
 			return nil, fmt.Errorf("failed to create static authorizer: %w", err)
 		}
 
-		delegated = append(delegated, staticAuthorizer)
+		authz = union.New(staticAuthorizer, authz)
```

Comment:
This is an odd change. Can you please write full unit test coverage of the original code (so a unit tests commit comes first) and then start moving authorizers around?